query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java

```diff
@@ -687,10 +706,107 @@ protected IckleFilterAndConverter createFilter(String queryString, Map<String, O
       return new IckleFilterAndConverter(queryString, namedParameters, matcherImplClass);
    }
 
+   public <E> CacheQuery<E> buildCacheQuery(org.apache.lucene.search.Query luceneQuery, IndexedQueryMode indexedQueryMode,
+                                            KeyTransformationHandler keyTransformationHandler,
+                                            TimeoutExceptionFactory timeoutExceptionFactory,
+                                            ExecutorService asyncExecutor,
+                                            Class<?>... classes) {
+      CacheQuery cacheQuery;
+      if (indexedQueryMode == IndexedQueryMode.BROADCAST) {
+         cacheQuery = new ClusteredCacheQueryImpl<>(luceneQuery, getSearchFactory(), asyncExecutor, cache,
+               keyTransformationHandler, classes);
+      } else {
+         cacheQuery = new CacheQueryImpl<>(luceneQuery, getSearchFactory(), cache, keyTransformationHandler,
+               timeoutExceptionFactory, classes);
+      }
+      return (CacheQuery<E>) cacheQuery;
+   }
+
+   public HsQueryRequest createHsQuery(String queryString, IndexedTypeMap<CustomTypeMetadata> metadata, Map<String, Object> nameParameters) {
+      IckleParsingResult<TypeMetadata> parsingResult = parse(queryString);
+      if (parsingResult.hasGroupingOrAggregations()) {
+         throw log.groupAggregationsNotSupported();
+      }
+      LuceneQueryParsingResult luceneParsingResult = transformParsingResult(parsingResult, nameParameters);
+      org.apache.lucene.search.Query luceneQuery = makeTypeQuery(luceneParsingResult.getQuery(), luceneParsingResult.getTargetEntityName());
+      SearchIntegrator searchFactory = getSearchFactory();
+      HSQuery hsQuery = metadata == null ? searchFactory.createHSQuery(luceneQuery) : searchFactory.createHSQuery(luceneQuery);
```

Comment:
I believe here you intended to write ```HSQuery hsQuery = metadata == null ? searchFactory.createHSQuery(luceneQuery) : searchFactory.createHSQuery(luceneQuery, metadata);```

Reply:
Fixed this here: https://github.com/infinispan/infinispan/commit/29e92eec857e16eae487353b89745b71891c390a